### DISCUSSION TITLE
Wrong handshake hash on read message for Transport

### DISCUSSION THREAD

**Discussion by kuzmordas** [sander/noise-kotlin#15] [General]
It seems that there are some small bug in `readMessage` method.
Please look on this [line](https://github.com/sander/noise-kotlin/blob/53f09dbaafda6eeb4480817888041a411cc6de67/src/main/kotlin/Handshake.kt#L105):
```
.let { State(Transport(it.first, it.second, symmetry.handshakeHash.digest), s.result) }
```
 I think there are should be this instead:
```
.let { State(Transport(it.first, it.second, s.value.symmetry.handshakeHash.digest), s.result) }
```

**Comment by sander** [comment]
Thanks, good catch! Fixed in https://github.com/sander/noise-kotlin/releases/tag/v0.7.0
